Bindings/Java/Matlab/Utilities/osimTableFromStruct.m

```diff
@@ -130,35 +119,21 @@
         % append the RowVectorofVec3's to the opensim table
         osimtable.appendRow(iRow-1, row);
         % set the time value for the appended row
-        timeColumn.set(iRow-1, structdata.(labelnames{timeIndex})(iRow));
-    end
-    
-else
-    % Else the data is in doubles, then build a osimtable of doubles.
-    % We build the osimtable row by row, filling a row vector from left to
-    % right and then appending the row to the OpenSim osimtable. 
-    row = RowVector(nLabels, NaN);
-
-    for iRow = 1 : nRows
-        % create and fill a row of data 
-        for iCol = indexArray
-            % get double from col and row index.
-            rowdata = structdata.(labelnames{iCol+1})(iRow,:);
-            % update the value in the row. 
-            row.set(iCol, rowdata);
-        end
-        % Append the row vector to the opensim table
-        osimtable.appendRow(iRow-1, row);
-        % set the time value
-        timeColumn.set(iRow-1, structdata.(labelnames{timeIndex})(iRow) );
+        timeColumn.set(iRow-1, s.(labels{tIndex})(iRow));
     end
 end
 
-% type change the DataTable to a TimeSeriesTable 
-if vec3table == true
-   timeseriesosimtable = TimeSeriesTableVec3(osimtable);
+%% Set the Time values
+DataTabletime = osimtable.getIndependentColumn();
+for i = 1 : nRows 
+      DataTabletime.set(i-1, time(i));
+end
+  
+%% Type change the DataTable to a TimeSeriesTable 
+if nCols == 1
+   timeseriesosimtable = TimeSeriesTable(osimtable);    
 else
-    timeseriesosimtable = TimeSeriesTable(osimtable);
+    timeseriesosimtable = TimeSeriesTableVec3(osimtable);
```

Comment:
Is there a reason you can't use a `TimeSeriesTable` to start with?

Reply:
I initially want to also support DataTable() type but I agree that this is not needed.